transport/http/roundtrip_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package http
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/yarpc/yarpc-go/transport"
+	test_trans "github.com/yarpc/yarpc-go/transport/testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRoundTrip(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	// start the inbound with a mock handler
+	h := test_trans.NewMockHandler(mockCtrl)
+	i := NewInbound(":0")
+	require.NoError(t, i.Start(h), "failed to Start()")
+	defer i.Stop()
+
+	request := &transport.Request{
+		Caller:    "testclient",
+		Service:   "mockservice",
+		Procedure: "hello",
+		Headers:   map[string]string{"Token": "1234"},
+		Body:      bytes.NewReader([]byte("world")),
+		TTL:       200 * time.Millisecond, // TODO use default
+	}
+
+	response := &transport.Response{
+		Headers: map[string]string{"status": "ok"},
+		Body:    ioutil.NopCloser(bytes.NewReader([]byte("hello, world"))),
+	}
+
+	// Need to build the response matcher before the response is consumed
+	// because otherwise we won't be able to read from response.Body again.
+	responseMatcher := test_trans.NewResponseMatcher(t, response)
+
+	addr := i.Addr().String() // this is something like "[::]:1234"
+	addr = addr[strings.LastIndexByte(addr, byte(':'))+1:]
```

Comment:
start your inbound on `127.0.0.1:0` instead to avoid this. you can then just do `fmt.Sprintf("http://%v/", addr)`